dpypelines/s3_tar_received.py

```diff
@@ -46,56 +50,24 @@ def start(s3_object_name: str):
             )
         ) from err
 
-    # Check for the existence of a configuration file
+    # get_dataset_id() currently empty - returns "not-specified"
+    # To be updated once we know where the dataset_id can be extracted from (not necessarily s3_object_name as suggested by argument name)
+    dataset_id = get_dataset_id(s3_object_name)
+
+    # Get config details for the given dataset_id
     try:
-        if not local_store.has_lone_file_matching(r"^pipeline-config.json$"):
-            de_notifier.failure()
-            msg = message.expected_local_file_missing(
-                "Pipeline config not found",
-                "pipeline-config.json",
-                "dataset_ingress_v1",
-                local_store,
-            )
-            raise ValueError(msg)
+        pipeline_config = get_pipeline_config(dataset_id, CONFIGURATION)
     except Exception as err:
         de_notifier.failure()
         raise Exception(
             message.unexpected_error(
-                "Error while checking for pipeline-config.json", err
+                "Failed to get pipeline configuration details", err
```

Comment:
I missed this previously but can you add some details to this error. so what the identifier was and what the keys in CONFIGURATION were, something like:

```
Failed to match dataset id "foo" to a configuration from ["bar", "baz"]
```
enough we can see what didnt match without digging.

Reply:
I've done this, but because I've removed the `CONFIGURATION` import, I had to add a new `config_keys` return value to `get_pipeline_config` so that the available keys can be included in the error message. If there's a better way, let me know.

Reply:
fair point, that'll do fine for now